plugins/auth-backend/src/identity/TokenFactory.ts

```diff
@@ -127,23 +128,31 @@ export class TokenFactory implements TokenIssuer {
     this.keyExpiry = moment().add(this.keyDurationSeconds, 'seconds');
     const promise = (async () => {
       // This generates a new signing key to be used to sign tokens until the next key rotation
-      const key = await JWK.generate('EC', 'P-256', {
-        use: 'sig',
-        kid: uuid(),
-        alg: 'ES256',
-      });
 
+      const key = await generateKeyPair('ES256');
+      const kid = uuid();
+      const jwk = await fromKeyLike(key.privateKey);
+
+      // @ts-ignore https://github.com/microsoft/TypeScript/issues/13195 -
+      // JOSE Library provides optional for most fields - and TS does not distinguish between missing/undefined.
+      // Because AnyJWK requires keys to have type "string", this throws a TypeError - though in practice, if the field
+      // is undefined, JOSE will not send it back as key.
+      const storedJwk: AnyJWK = {
```

Comment:
Would prefer casts over using `@ts-ignore`, as it's easy to hide other type errors by mistake. I think in this case it might be enough to use

```ts
const storedJwk = {
...
} as AnyJWK
```